src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Utf8Parser/Utf8Parser.Integer.Signed.cs

```diff
@@ -127,27 +127,27 @@ public static bool TryParse(ReadOnlySpan<byte> source, out short value, out int
         /// </exceptions>
         public static bool TryParse(ReadOnlySpan<byte> source, out int value, out int bytesConsumed, char standardFormat = default)
         {
-            switch (standardFormat)
-            {
-                case default(char):
-                case 'g':
-                case 'G':
-                case 'd':
-                case 'D':
-                    return TryParseInt32D(source, out value, out bytesConsumed);
+            int standardFormatLowercase;
 
-                case 'n':
-                case 'N':
-                    return TryParseInt32N(source, out value, out bytesConsumed);
+            if (standardFormat == default
+                || ((standardFormatLowercase = standardFormat | 0x20) == 'g')
+                || (standardFormatLowercase == 'd'))
+            {
+                return TryParseInt32D(source, out value, out bytesConsumed);
+            }
 
-                case 'x':
-                case 'X':
-                    value = default;
-                    return TryParseUInt32X(source, out Unsafe.As<int, uint>(ref value), out bytesConsumed);
+            if (standardFormatLowercase == 'n')
+            {
+                return TryParseInt32N(source, out value, out bytesConsumed);
+            }
 
-                default:
-                    return ParserHelpers.TryParseThrowFormatException(out value, out bytesConsumed);
+            if (standardFormatLowercase != 'x')
+            {
+                ThrowHelper.ThrowFormatException_BadFormatSpecifier();
             }
+
+            value = default;
+            return TryParseUInt32X(source, out Unsafe.As<int, uint>(ref value), out bytesConsumed);
```

Comment:
after moving from the previous `switch` to this `if` statement it became much harder to read. Did we get any noticeable performance gain by doing it?